### PR TITLE
COMP: Use C99 int types over libtiff's

### PR DESCRIPTION
**Pull request by blowekamp**
The libtiff int types are depricated in 4.3.0 and produce a warning.

## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

File: Modules/IO/LSM/src/itkLSMImageIO.cxx [hunk 1]
```diff
@@ -273,8 +273,8 @@ LSMImageIO::Write(const void * buffer)
     return;
   }
 
-  uint32 w = width;
-  uint32 h = height;
+  uint2_t w = width;
```
Comment (dzenanz) [inline Modules/IO/LSM/src/itkLSMImageIO.cxx:276]: `uint32_t`. Compiler has eagle eyes 🖥️🦅👁️ 😄 

Similar mistake appears in a few more places, e.g.
`Modules\IO\TIFF\src\itkTIFFReaderInternal.cxx(149): error C2065: 'int2_t': undeclared identifier`

**Comment by Leengit** [comment]
You can fix everything from some topLevelDirectory with:
```bash
egrep -Rl '[^_A-Za-z]u?int(8|16|32|64)[^_]' topLevelDirectory | egrep '\.(h|hxx|hpp|c|cc|cxx|cpp)$' |
    xargs sed -i -r -e 's/([^_A-Za-z]u?int(8|16|32|64))([^_])/\1_t\3/g'
```

**Comment by blowekamp** [comment]
> You can fix everything from some topLevelDirectory with:
> 
> ```shell
> egrep -Rl '[^_A-Za-z]u?int(8|16|32|64)[^_]' topLevelDirectory | egrep '\.(h|hxx|hpp|c|cc|cxx|cpp)$' |
>     xargs sed -i -r -e 's/([^_A-Za-z]u?int(8|16|32|64))([^_])/\1_t\3/g'
> ```

I used `sed -i .tmp -E "s/int([0-9])+ /int\1_t /" `, but I believe should be corrected to: `sed -i .tmp -E "s/int([0-9]+) /int\1_t /"

**Comment by Leengit** [comment]
> > You can fix everything from some topLevelDirectory with:
> > ```shell
> > egrep -Rl '[^_A-Za-z]u?int(8|16|32|64)[^_]' topLevelDirectory | egrep '\.(h|hxx|hpp|c|cc|cxx|cpp)$' |
> >     xargs sed -i -r -e 's/([^_A-Za-z]u?int(8|16|32|64))([^_])/\1_t\3/g'
> > ```
> 
> I used `sed -i .tmp -E "s/int([0-9])+ /int\1_t /" `, but I believe should be corrected to: `sed -i .tmp -E "s/int([0-9]+) /int\1_t /"

I tried to handle cases like `__int32` (avoid) and `MyTemplate<int32>` (convert).

**Comment by Leengit** [comment]
> ```shell
> egrep -Rl '[^_A-Za-z]u?int(8|16|32|64)[^_]' topLevelDirectory | egrep '\.(h|hxx|hpp|c|cc|cxx|cpp)$' |
>     xargs sed -i -r -e 's/([^_A-Za-z]u?int(8|16|32|64))([^_])/\1_t\3/g'
> ```

But I didn't handle the cases where (1) the part before `u?int` is the start of the line nor (2) the part after the number of bits is the end of the line, nor (3) both.  So we need three more commands similar to the above ... or a fancier single command.

**Review by N-Dekker — APPROVED** [review]
We may still consider the corresponding `std` integer types from standard C++ header file `<cstdint>` (possibly after some more discussion). But I think this PR is already a significant improvement. Approved!

**Review by dzenanz — APPROVED** [review]

**Pull request merged**